src/main/java/uk/gov/hmcts/reform/et/syaapi/service/CaseRoleManagementService.java

```diff
@@ -0,0 +1,245 @@
+import static uk.gov.hmcts.reform.et.syaapi.constants.CaseRoleManagementConstants.MODIFY_CASE_ROLE_EMPTY_REQUEST;
+import static uk.gov.hmcts.reform.et.syaapi.constants.CaseRoleManagementConstants.MODIFY_CASE_ROLE_POST_WORDING;
+import static uk.gov.hmcts.reform.et.syaapi.constants.CaseRoleManagementConstants.MODIFY_CASE_ROLE_PRE_WORDING;
+import static uk.gov.hmcts.reform.et.syaapi.constants.EtSyaConstants.ENGLAND_CASE_TYPE;
+import static uk.gov.hmcts.reform.et.syaapi.constants.EtSyaConstants.SCOTLAND_CASE_TYPE;
+import static uk.gov.hmcts.reform.et.syaapi.enums.CaseEvent.UPDATE_CASE_SUBMITTED;
+import static uk.gov.hmcts.reform.et.syaapi.service.utils.CaseRoleManagementServiceUtil.buildHeaders;
+import static uk.gov.hmcts.reform.et.syaapi.service.utils.CaseRoleManagementServiceUtil.generateCaseDataByUserInfoCaseDetails;
+import static uk.gov.hmcts.reform.et.syaapi.service.utils.CaseRoleManagementServiceUtil.getHttpMethodByModificationType;
+
+/**
+ * Provides read and write access to cases stored by ET.
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CaseRoleManagementService {
+
+    private String roleList;
+
+    private final AdminUserService adminUserService;
+    private final RestTemplate restTemplate;
+    private final AuthTokenGenerator authTokenGenerator;
+    private final CoreCaseDataApi ccdApi;
+    private final IdamClient idamClient;
+    private final CaseService caseService;
+    private final CaseDetailsConverter caseDetailsConverter;
+
+    @Value("${core_case_data.api.url}")
+    private String ccdDataStoreUrl;
+
+    /**
+     * Gets case with the user entered details, caseId, respondentName, claimantFirstNames and claimantSurname.
+     * Returns null when case not found. It searches for the cases with administrator user to find if the case
+     * exists with the given parameters. Also makes a security check if the user entered valid values.
+     * @param findCaseForRoleModificationRequest It has the values caseId, respondentName, claimantFirstNames and
+     *                                           claimantSurname values given by the respondent.
+     * @return null if no case is found, CaseDetails if any case is found in both scotland and england wales case
+     *              types.
+     */
+    public CaseDetails findCaseForRoleModification(
+        FindCaseForRoleModificationRequest findCaseForRoleModificationRequest) {
+        log.info(
+            "Trying to receive case for role modification. Submission Reference: {}",
+            findCaseForRoleModificationRequest.getCaseSubmissionReference()
+        );
+        String adminUserToken = adminUserService.getAdminUserToken();
+
+        List<CaseDetails> englandCases = Optional.ofNullable(ccdApi.searchCases(
+            adminUserToken,
+            authTokenGenerator.generate(),
+            ENGLAND_CASE_TYPE,
+            ElasticSearchQueryBuilder.buildElasticSearchQueryForRoleModification(findCaseForRoleModificationRequest)
```

Comment:
Does it make sense to store this as a variable rather than calling the method twice (again at line 102)

Reply:
Created a local variable and assigned to that 👍